src/security/Handshake.cc

```diff
@@ -547,26 +551,29 @@ Security::HandshakeParser::ParseCertificate(const SBuf &raw, Security::CertPoint
     Must(x509Pos == x509Start + raw.length()); // no leftovers
 #else
     // workaround GCC -O3 error with unused variables. see bug 4663.
-    pCert = nullptr;
-    debugs(83, 2, "TLS parsing is not supported without OpenSSL. " << raw);
```

Comment:
I do not think you should because this code must not run after your changes. Adding debugs() to the code that must not run does not make sense to me. I suggest:

* If we can implement the [ideal solution](https://github.com/squid-cache/squid/pull/42#discussion_r132222293), implement it.
* Otherwise, add an assert to use `raw`. For example:
```
#else
assert(false); // this code should never be reached
pCert = Security::CertPointer(nullptr); // avoid warnings about uninitialized pCert; XXX: Fix CertPoint declaration.
assert(&raw != nullptr); // avoid warnings about unused method parameter; TODO: Add a SimulateUse() function.
#endif
```